src/view/element.js

```diff
@@ -123,6 +123,17 @@ export default class Element extends Node {
 	}
 
 	/**
+	 * Length of this view element in model if it is mapped. This value can be overwritten in classes extending
+	 * {@link engine.view.Element Element} to provide correct position mapping in {@link engine.conversion.Mapper Mapper}.
+	 *
+	 * @readonly
+	 * @type {Number}
+	 */
+	get modelLength() {
```

Comment:
Okay so I don't like this one too. I was supposed to think more of it but I forgot. I don't like mentioning mapping and model in the view classes and I am all for doing it some other way.

Unfortunately, we still need that `modelLength` thing, even if we use callbacks in `Mapper`. It is needed for "default" mapper algorithm, because we don't fire custom callbacks "deeply" (after each position change when looking for correct position).

I set it to `1` and implemented it so there is a track in documentation that such thing is available and you can overwrite it.

Since this is bad, here is my other proposal:
`Mapper.registerViewElementLength( ViewElementClass, lengthCallback );`

This would map custom view element classes with functions that will calculate and return length of given view element in model. This could be either instance method or static method. BTW. I was thinking whether make position mapping callback per instance or static. Per instance = better flexibility, but it often ends up with just adding same thing in multiple places instead of one.

So, anyway, right now, in `List` feature I have:
```javascript
class ViewListItemElement extends ViewContainerElement {
	// ... cut ...

	get modelLength() {
		let length = 1;

		for ( let child of this.getChildren() ) {
			if ( child.name == 'ul' || child.name == 'ol' ) {
				for ( let item of child.getChildren() ) {
					length += item.modelLength;
				}
			}
		}

		return length;
	}
}
```
It will be changed to:
```javascript
mapper.registerViewElementLength( ViewListItemElement, ( element ) => {
	let length = 1;

	for ( let child of this.getChildren() ) {
		if ( child.name == 'ul' || child.name == 'ol' ) {
			for ( let item of child.getChildren() ) {
				length += item.modelLength;
			}
		}
	}

	return length;
} );
```
WDYT?